#  Chi-Squared Test

## Goodness-of-Fit Test
#### “does my distribution of numbers come from the distribution I expect?”

>The null and alternative hypotheses for a Chi-Square Goodness of Fit Test are as follows:
>
>- H0: variable is from the expected distribution
>- H1: A variable differs from the expected distribution

Criteria
>if chi_square_ value > critical value, the null hypothesis is rejected. if chi_square_ value <= critical value, the null hypothesis is accepted.\
>
>if p-value greater than or equal to 0.05 is not statistically significant and suggests strong support for the null hypothesis and alternative hypothesis is rejected.

In [11]:
# importing packages
import scipy.stats as stats
import numpy as np

# no of hours a student studies
# in a week vs expected no of hours
observed_data = [8, 6, 10, 7, 8, 11, 9]
expected_data = [9, 8, 11, 8, 10, 7, 6]



# Chi-Square Goodness of Fit Test
chi_square_test_statistic, p_value = stats.chisquare(
	observed_data, expected_data)

# Round the chi square test statistic and p value
chi_square_test_statistic=round(chi_square_test_statistic,2)
p_value=round(p_value,2)

# chi square test statistic and p value
print('chi_square_test_statistic is : ' +
	str(chi_square_test_statistic))
print('p_value : ' + str(round(p_value,2)))

# find Chi-Square critical value
critical_value = round(stats.chi2.ppf(1 - 0.05, df=6),2)
print('Critical Value: ' + str(critical_value))

# chi square calculation with formula
chi_square_test_statistic1 = 0
for i in range(len(observed_data)):
    chi_square_test_statistic1 = chi_square_test_statistic1 + \
        (np.square(observed_data[i]-expected_data[i]))/expected_data[i]
  
  
print('chi square value determined by formula : ' +
      str(chi_square_test_statistic1))


chi_square_test_statistic is : 5.01
p_value : 0.54
Critical Value: 12.59
chi square value determined by formula : 5.0127344877344875


NOTE:
>H0: (variable is from the expected distribution) is accepted because p_value : 0.542180861413329 hence p-value greater than or equal to 0.05 is not statistically significant and suggests strong support for the null hypothesis. This signifies that the null hypothesis is kept and the alternative hypothesis is rejected. There is 54% chance
>
>chi square test statistic(5.01) <= critical value(12.59), This also signifies that the null hypothesis is kept and the alternative hypothesis is rejected.